src/IECoreScene/CurvesAlgo.cpp

```diff
@@ -200,7 +200,17 @@ struct  CurvesUniformToVarying
 		for( size_t i = 0; i < numCurves; ++i, ++srcIt )
 		{
 			Canceller::check( m_canceller );
-			for( size_t j = 0; j < m_curves->numSegments( i ) + 1; ++j )
+			size_t varyingSize;
+			if( m_curves->periodic() )
+			{
+				varyingSize = m_curves->numSegments( i );
+			}
+			else
+			{
+				varyingSize = m_curves->numSegments( i ) + 1;
+			}
+
+			for( size_t j = 0; j < varyingSize; ++j )
```

Comment:
Could be a one liner I think :

`const size_t varyingSize = m_curves->variableSize( PrimitiveVariable::Varying, i )`.

Reply:
Same applies to the other changes below...